lib/engine/route.rb

```diff
@@ -27,6 +27,7 @@ def initialize(game, phase, train, **opts)
     end
 
     def clear_cache!
+      connection_data
```

Comment:
maybe only do this if @connection_data already?

Reply:
Actually, it needs to happen if !@connection_data. That's the case where blowing away @connection_hexes prevents reconstruction of @connection_data

Reply:
Changed, per our discussion.